projects/extension/sql/idempotent/012-vectorizer-int.sql

```diff
@@ -465,76 +461,147 @@ $func$
+
+        elsif (TG_LEVEL = 'STATEMENT') then
+            if (TG_OP = 'TRUNCATE') then
+                execute format('truncate table %I.%I', '$TARGET_SCHEMA$', '$TARGET_TABLE$');
+            end if;
+            return null;
+        end if;
+        
         return null;
     end;
-    $plpgsql$ language plpgsql volatile parallel safe security definer
-    set search_path to pg_catalog, pg_temp
-    $sql$
-    , queue_schema, trigger_name
-    , queue_schema, queue_table
-    , (
-        select pg_catalog.string_agg(pg_catalog.format('%I', x.attname), ', ' order by x.attnum)
-        from pg_catalog.jsonb_to_recordset(source_pk) x(attnum int, attname name)
-      )
-    , (
-        select pg_catalog.string_agg(pg_catalog.format('new.%I', x.attname), ', ' order by x.attnum)
-        from pg_catalog.jsonb_to_recordset(source_pk) x(attnum int, attname name)
-      )
-    ) into strict _sql
-    ;
+    $def$;
+
+    -- Replace placeholders
+    _func_def := replace(_func_def, '$DELETE_STATEMENT$', _delete_statement);
+    _func_def := replace(_func_def, '$PK_CHANGE_CHECK$', _pk_change_check);
+    _func_def := replace(_func_def, '$QUEUE_SCHEMA$', quote_ident(queue_schema));
+    _func_def := replace(_func_def, '$QUEUE_TABLE$', quote_ident(queue_table));
+    _func_def := replace(_func_def, '$PK_COLUMNS$', _pk_columns);
+    _func_def := replace(_func_def, '$PK_VALUES$', _pk_values);
+    _func_def := replace(_func_def, '$TARGET_SCHEMA$', quote_ident(target_schema));
+    _func_def := replace(_func_def, '$TARGET_TABLE$', quote_ident(target_table));
+
+    return _func_def;
+end;
+$func$ language plpgsql stable security invoker
+set search_path to pg_catalog, pg_temp;
+
+-------------------------------------------------------------------------------
+-- _vectorizer_create_source_trigger
+create or replace function ai._vectorizer_create_source_trigger
+( trigger_name pg_catalog.name     -- Name for the trigger
+, queue_schema pg_catalog.name     -- Schema containing the queue table
+, queue_table pg_catalog.name      -- Table that will store queued items
+, source_schema pg_catalog.name    -- Schema containing the watched table
+, source_table pg_catalog.name     -- Table being watched for changes
+, target_schema pg_catalog.name    -- Schema containing the target table for deletions
+, target_table pg_catalog.name     -- Table where corresponding rows should be deleted
+, source_pk pg_catalog.jsonb       -- JSON describing primary key columns to track
+, original_owner pg_catalog.name default null   -- Role that should own the trigger
+) returns void as
+$func$
+declare
+    _sql pg_catalog.text;
+begin
+    
+    execute format(
+        'create or replace function ai.%I() returns trigger as $trigger_def$ %s $trigger_def$ language plpgsql volatile parallel safe security definer',
+        trigger_name,
+        ai._vectorizer_build_trigger_definition(queue_schema, queue_table, target_schema, target_table, source_pk)
+    );
```

Comment:
```suggestion
    execute format
    ( $sql$
    create function %I.%I() returns trigger 
    as $trigger_def$ 
    %s
    $trigger_def$ language plpgsql volatile parallel safe security definer 
    set search_path to pg_catalog, pg_temp
    $sql$
    , queue_schema
    , trigger_name
    , ai._vectorizer_build_trigger_definition(queue_schema, queue_table, target_schema, target_table, source_pk)
    );
```

1. We want to error if the function already exists. We were doing that before.
2. The function should go into the same schema as the queue table, as before.
3. Set the search_path explicitly on the function
4. very nitpicky formatting, sorry.